modules/metastorage/src/test/java/org/apache/ignite/internal/metastorage/impl/MetaStorageDeployWatchesCorrectnessTest.java

```diff
@@ -0,0 +1,95 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.Set;
+import org.apache.ignite.internal.cluster.management.ClusterManagementGroupManager;
+import org.apache.ignite.internal.cluster.management.topology.api.LogicalTopologyService;
+import org.apache.ignite.internal.hlc.HybridClock;
+import org.apache.ignite.internal.hlc.HybridClockImpl;
+import org.apache.ignite.internal.metastorage.MetaStorageManager;
+import org.apache.ignite.internal.metastorage.server.SimpleInMemoryKeyValueStorage;
+import org.apache.ignite.internal.raft.RaftManager;
+import org.apache.ignite.internal.testframework.IgniteAbstractTest;
+import org.apache.ignite.internal.vault.VaultManager;
+import org.apache.ignite.internal.vault.inmemory.InMemoryVaultService;
+import org.apache.ignite.lang.NodeStoppingException;
+import org.apache.ignite.network.ClusterService;
+import org.junit.jupiter.api.Test;
+
+/**
+ * There are tests of correctness invocation {@link MetaStorageManager#deployWatches()}.
+ */
+public class MetaStorageDeployWatchesCorrectnessTest extends IgniteAbstractTest {
+    @Test
+    public void tesMetaStorageManager() throws Exception {
+        HybridClock clock = new HybridClockImpl();
+        String mcNodeName = "mc-node-1";
+
+        ClusterManagementGroupManager cmgManager = mock(ClusterManagementGroupManager.class);
+        ClusterService clusterService = mock(ClusterService.class);
+        LogicalTopologyService logicalTopologyService = mock(LogicalTopologyService.class);
+        RaftManager raftManager = mock(RaftManager.class);
+
+        when(cmgManager.metaStorageNodes()).thenReturn(completedFuture(Set.of(mcNodeName)));
+        when(clusterService.nodeName()).thenReturn(mcNodeName);
+        when(raftManager.startRaftGroupNodeAndWaitNodeReadyFuture(any(), any(), any(), any(), any())).thenReturn(completedFuture(null));
+
+        var metastore = new MetaStorageManagerImpl(
+                new VaultManager(new InMemoryVaultService()),
+                clusterService,
+                cmgManager,
+                logicalTopologyService,
+                raftManager,
+                new SimpleInMemoryKeyValueStorage(mcNodeName),
+                clock
+        );
+
+        checkCorrectness(metastore);
+    }
+
+    @Test
+    public void tesStandaloneMetaStorageManager() throws Exception {
+        var metastore = StandaloneMetaStorageManager.create(new VaultManager(new InMemoryVaultService()));
+
+        checkCorrectness(metastore);
+    }
+
+    /**
+     * Invokes {@link MetaStorageManager#deployWatches()} and checks result.
+     *
+     * @param metastore Meta storage.
+     * @throws NodeStoppingException If failed.
+     */
+    private static void checkCorrectness(MetaStorageManager metastore) throws NodeStoppingException {
+        var deployWatchesFut = metastore.deployWatches();
+
+        assertFalse(deployWatchesFut.isDone());
+
+        metastore.start();
```

Comment:
metastore.start() is  async thus `deployWatchesFut.isDone()` isn't guaranteed without some awaiting logic. Please use 
`org.apache.ignite.internal.testframework.matchers.CompletableFutureMatcher#willSucceedFast` or similar instead.

Reply:
It is just a unit test, I know the behavior here.
I am sure we will invoke the listener in the same thread where the start future is completed.

Reply:
Why should we exposes such implementation details (e.g. the one that there's no executors or similar inside metaStorage.start) instead of writing guaranteed code based on willSucceedFast?